lib/services/local/saml_idp_service_provider.go

```diff
@@ -18,15 +18,21 @@ package local
 
 import (
 	"context"
+	"time"
 
+	"github.com/crewjam/saml/samlsp"
 	"github.com/gravitational/trace"
 
 	"github.com/gravitational/teleport/api/types"
 	"github.com/gravitational/teleport/lib/backend"
 	"github.com/gravitational/teleport/lib/services"
 )
 
-const samlIDPServiceProviderMaxPageSize = 200
+const (
+	samlIDPServiceProviderModifyLock    = "saml_idp_service_provider_modify_lock"
+	samlIDPServiceProviderModifyLockTTL = time.Second * 5
+	samlIDPServiceProviderMaxPageSize   = 200
+)
```

Comment:
I'd move this together with `samlIDPServiceProviderPrefix` (or move that one up here).